java/src/org/certificatetransparency/ctlog/serialization/Deserializer.java

```diff
@@ -168,18 +169,17 @@ public static TimestampedEntry parseTimestampedEntry(InputStream in) {
       int length = (int) readNumber(in, 3);
       signedEntry.x509 = readFixedLength(in, length);
     } else if (entryType == Ct.LogEntryType.PRECERT_ENTRY_VALUE) {
-      Ct.PreCert.Builder preCertBuilder = Ct.PreCert.newBuilder();
+      PreCert preCert = new PreCert();
 
-      byte[] arr = readFixedLength(in, 32);
-      preCertBuilder.setIssuerKeyHash(ByteString.copyFrom(arr));
+      preCert.issuerKeyHash = readFixedLength(in, 32);
 
       // set tbs certificate
-      arr = readFixedLength(in, 2);
+      byte[] arr = readFixedLength(in, 2);
```

Comment:
Hmm, that method seems to raise an exception if the underlying stream doesn't support seek, and the `readFixedLength` also handles short reads, which isn't the worst, but is also more work here...